graphql-language/src/parser.rs

```diff
@@ -0,0 +1,363 @@
+#![allow(dead_code, unused_variables)]
+use ast::*;
+use lexer::Punctuator::*;
+use lexer::Token::{
+  Eof,
+  Punctuator,
+  Name as TokenName,
```

Comment:
I've come around to think that aliasing in rust is a bad idea. Either use namespaces (`Token::Name`) or just use its full name `Name`. In this case I would suggest imports which look like the following:

```rust
use ast;
use lexer::Punctuator::*;
use lexer::Token::*;
use lexer::{Lexer, Token};
use source::Source;
use std::iter::Peekable;
```

So you would differentiate between the token name and the AST name like `Name` and `ast::Name` respectively.